test/framework/src/main/java/org/elasticsearch/test/jar/JarUtils.java

```diff
@@ -69,6 +72,19 @@ public static void createJarWithEntries(Path jarfile, Map<String, byte[]> entrie
         }
     }
 
+    /**
+     * Creates a jar file with the given entries.
+     *
+     * @param jarfile the jar file path
+     * @param entries map of entries to add; jar entry name to String contents
+     * @param charset the charset used to convert the entry values to bytes
+     * @throws IOException if an I/O error occurs
+     */
+    public static void createJarWithEntries(Path jarfile, Map<String, String> entries, Charset charset) throws IOException {
+        var map = entries.entrySet().stream().collect(toUnmodifiableMap(Map.Entry::getKey, v -> v.getValue().getBytes(charset)));
```

Comment:
When would we ever want to have non-utf8? Could we just have this method assume that and remove the Charset param?

Reply:
Done.